src/app/view/tools/insertTextTool.js

```diff
@@ -9,6 +9,9 @@ import PositionTool from "./positionTool";
 import Params from "../../core/params";
 
 export default class InsertTextTool extends MouseTool {
+    
+    text;
```

Comment:
this should be `private _text: string;`

Reply:
In order to make this private, I need to create a Typescript file. This has presented several problems.

First, there are functions (up, down, move) which 'MouseTool' defines in its constructor and take no argument, but these functions take an argument in 'insertTextTool'. As far as I can determine, Typescript does not allow an inherited function to have its arguments changed in this way. I've address this by adding an (optional) argument to the parent class declarations; I assume this is acceptable.

Second, I need to assign a type to the 'event' arguments throughout the class. I suspect these are 'Event' objects from Event.ts; however, the filepath for this is not allowing anything but the absolute filepath. So I am seeking confirmation that these 'event's are indeed from Event.ts.

Lastly, insertTextTool has numerous uses of 'Registry' members which are generating errors due to them being set to null in Registry.ts. I am unsure whether to try removing the uses of Registry or modify the members in Registry

Reply:
I forgot that this is a js file